clients/python/test/e2e/ci/setup_e2e_pytest.py

```diff
@@ -0,0 +1,121 @@
+import toml
+from pathlib import Path
+import pandas as pd
+from urllib.parse import urlparse, ParseResult
+from typing import List, Dict, Any, Union
+import typer
+from packaging import version
+import json
+from _utils import E2eScriptFailure, E2eExitCodes
+import warnings
+
+# keys to be found in input dicts
+surl: str = "OSPARC_API_HOST"
+skey: str = "OSPARC_API_KEY"
+ssecret: str = "OSPARC_API_SECRET"
+skeys: List[str] = [surl, skey, ssecret]
+
+crepo: str = "OSPARC_CLIENT_REPO"
+cbranch: str = "OSPARC_CLIENT_BRANCH"
+cversion: str = "OSPARC_CLIENT_VERSION"
+ckeys: List[str] = [crepo, cbranch, cversion]
+
+
+def main(client_config: str, server_config: str) -> bool:
+    """
+    Generates a toml configuration file pytest e2e tests
+
+    exceptions:
+    -----------
+        A typer.Exit(code=100) is raised if a failure is encountered
+
+    returns:
+    --------
+        A bool indicating whether or not the (client, server) pair are compatible
+    """
+    # read in data
+    ccfg = json.loads(client_config)
```

Comment:
STYLE: do not be shy with the name of variables :-)   I would only use sort variable names e.g. in loop indexes ... where their name sometimes is cluttering more than clarifying. In this case, these variables have client and server configuration , they are used along a large extensio of the code and are important enough to deserve a good human readable name :-)